mlflow/store/model_registry/sqlalchemy_store.py

```diff
@@ -738,47 +815,25 @@ def search_model_versions(self, filter_string):
         :return: PagedList of :py:class:`mlflow.entities.model_registry.ModelVersion`
                  objects.
         """
-        parsed_filter = SearchUtils.parse_filter_for_model_versions(filter_string)
-        if len(parsed_filter) == 0:
-            conditions = []
-        elif len(parsed_filter) == 1:
-            filter_dict = parsed_filter[0]
-            if filter_dict["comparator"] not in SearchUtils.VALID_MODEL_VERSIONS_SEARCH_COMPARATORS:
-                raise MlflowException(
-                    "Model Registry search filter only supports the equality(=) "
-                    "comparator and the IN operator "
-                    "for the run_id parameter. Input filter string: %s" % filter_string,
-                    error_code=INVALID_PARAMETER_VALUE,
-                )
-            if filter_dict["key"] == "name":
-                conditions = [SqlModelVersion.name == filter_dict["value"]]
-            elif filter_dict["key"] == "source_path":
-                conditions = [SqlModelVersion.source == filter_dict["value"]]
-            elif filter_dict["key"] == "run_id":
-                if filter_dict["comparator"] == "IN":
-                    conditions = [SqlModelVersion.run_id.in_(filter_dict["value"])]
-                else:
-                    conditions = [SqlModelVersion.run_id == filter_dict["value"]]
-            else:
-                raise MlflowException(
-                    "Invalid filter string: %s" % filter_string, error_code=INVALID_PARAMETER_VALUE
-                )
-        else:
-            raise MlflowException(
-                "Model Registry expects filter to be one of "
-                "\"name = '<model_name>'\" or "
-                "\"source_path = '<source_path>'\" "
-                'or "run_id = \'<run_id>\' or "run_id IN (<run_ids>)".'
-                "Input filter string: %s. " % filter_string,
-                error_code=INVALID_PARAMETER_VALUE,
-            )
+        parsed_filters = SearchModelUtils.parse_search_filter(filter_string)
+
+        attribute_filters, non_attribute_filters = self._get_search_model_versions_filter_clauses(
+            parsed_filters, self.engine.dialect.name
+        )
```

Comment:
We're going to revert changes for `search_model_versions` in this PR, right?

Reply:
We need to support tag searching for both search_registered_model and search_model_versions as @dbczumar said.

Reply:
So we'll update the Databricks backend later?

Reply:
Databricks backend already supports tab-based model version search,
but Databricks backend haven't support  tab-based search_registered_model which we need update later.

Reply:
Ok, I thought we were going to update `search_registered_models` and `search_model_versions` in the following order:

1. First, update `search_registered_models` (because Databricks backend support tag-based registered model search)
2. Then, update Databricks backend to support tab-based model version search.
3. Then, `search_model_versions` to support tags.

Reply:
ok.
Or I can file databricks side PR and merge it first, then merge this OSS PR, this avoid to split the PR into 2 parts.

Reply:
Splitting a PR sounds ok to me.

Reply:
Let's continue with reviewing / merging this PR now that we have a PR on the Databricks side :). Keeping the PRs together is fine.